pkg/landscaper/controllers/deployitem/actuator.go

```diff
@@ -0,0 +1,126 @@
+// SPDX-FileCopyrightText: 2020 SAP SE or an SAP affiliate company and Gardener contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package execution
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/inject"
+
+	lsv1alpha1 "github.com/gardener/landscaper/apis/core/v1alpha1"
+	lsv1alpha1helper "github.com/gardener/landscaper/apis/core/v1alpha1/helper"
+	"github.com/gardener/landscaper/pkg/landscaper/deployitem"
+	"github.com/gardener/landscaper/pkg/landscaper/operation"
+)
+
+func NewActuator() (reconcile.Reconciler, error) {
+	return &actuator{}, nil
+}
+
+type actuator struct {
+	log    logr.Logger
+	c      client.Client
+	scheme *runtime.Scheme
+}
+
+const (
+	defaultPickupTimeout   = int64(300)         // default pickup timeout in seconds
+	pickupTimeoutReason    = "PickupTimeout"    // for error messages
+	pickupTimeoutOperation = "WaitingForPickup" // for error messages
+)
+
+var _ inject.Client = &actuator{}
+
+var _ inject.Logger = &actuator{}
+
+var _ inject.Scheme = &actuator{}
+
+// InjectClients injects the current kubernetes client into the actuator
+func (a *actuator) InjectClient(c client.Client) error {
+	a.c = c
+	return nil
+}
+
+// InjectLogger injects a logging instance into the actuator
+func (a *actuator) InjectLogger(log logr.Logger) error {
+	a.log = log
+	return nil
+}
+
+// InjectScheme injects the current scheme into the actuator
+func (a *actuator) InjectScheme(scheme *runtime.Scheme) error {
+	a.scheme = scheme
+	return nil
+}
```

Comment:
inject methods are deprecated by the controller library we should not use them in the new controllers.
instead just add them during the initialization of the controller